src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1048,21 +1048,24 @@ private void RunConnectRequestHandlerTask(ConnectParams connectParams)
         /// <summary>
         /// Handle requests to list databases on the current server
         /// </summary>
-        protected async Task HandleListDatabasesRequest(
+        protected Task HandleListDatabasesRequest(
             ListDatabasesParams listDatabasesParams,
             RequestContext<ListDatabasesResponse> requestContext)
         {
-            Logger.Write(TraceEventType.Verbose, "ListDatabasesRequest");
-
-            try
-            {
-                ListDatabasesResponse result = ListDatabases(listDatabasesParams);
-                await requestContext.SendResult(result);
-            }
-            catch (Exception ex)
+            Task.Run(async () =>
             {
-                await requestContext.SendError(ex.ToString());
-            }
+                Logger.Write(TraceEventType.Verbose, "ListDatabasesRequest");
+                try
+                {
+                    ListDatabasesResponse result = ListDatabases(listDatabasesParams);
+                    await requestContext.SendResult(result);
+                }
+                catch (Exception ex)
+                {
+                    await requestContext.SendError(ex.ToString());
+                }
+            });
+            return Task.CompletedTask;

```

Comment:
Why are you returning Task.CompletedTask instead of the return value of Task.Run?

Reply:
If I return the task.run, it will still block the main thread.

![image](https://user-images.githubusercontent.com/13777222/117379726-f2608b80-ae8c-11eb-96b2-bbb6c8f4eac5.png)
https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs#L304

 I haven't reviewed all the request handlers, there might be request types that we actually want to block the main thread, maybe that is why it is designed to return a task.

Reply:
for example: https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs#L151
looks like it is on a different thread, but it is actually blocking during my testing.

Reply:
as mentioned in the pr description, I've created a task to review all the handlers after may release.